leasing/report/utils.py

```diff
@@ -3,3 +3,35 @@ def dictfetchall(cursor):
     """Return all rows from a cursor as a dict"""
     columns = [col[0] for col in cursor.description]
     return [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+
+# Gaps in billing periods helpers
+
+
+def set_active_rent_period(row, today, start, end):
```

Comment:
If `row` is a lease like the docstring suggests, it should maybe be named as lease, also would be great to type it as Lease too. If it is something else, typing would be preferable in any case for all the arguments.
Also looking at the function arguments, if e.g. `start` is `start_date` like in the row, it would be better to keep that name in my opinion.

Reply:
The name of the function is misleading, as it is not setting anything, rather getting

Reply:
The row is not a lease, it is custom row of data, and I found it difficult to name. I chose billing_period_data_row. It's not a good name, but the best that I could come up with.

Reply:
I would like to declare the type for billing_period_data_row, but I am not sure how to do it properly in Python. The best that I could come up with was to define a TypeDict-based class like this, but I am not sure how to use it.

```code
class BillingPeriodDataRow(TypeDict):
    lease_identifier: str
    lease_id: int
    start_date: date
    end_date: date
    rent_id: int
    rent_start_date: date
    rent_end_date: date
    invoice_id: int
    billing_period_start_date: date
    billing_period_end_date: date
```